Amplify/Categories/DataStore/Model/DataType/Date+DateScalar.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// An extension that makes the `Date` struct conform with the `DataScalar` protocol.
+/// When used in persistence operations, the granularity of the different date representations
+/// is set by using different scalar types: `Date`, `DateTime` and `Time`.
+///
+/// In those scenarios, the standard `Date` is formatted to ISO-8601 without the time.
+/// When the full date information is required, use `DateTime` instead.
+///
+extension Date: DateScalar, DateUnitOperable {
```

Comment:
My read of this code is that this extends the Foundation `Date` type with new properties and init methods, but does not otherwise alter its functionality, so that if a client app is relying on particular date behaviors, they're not going to be affected by this at all. Is that correct?

However, as a type that represents just the calendar date (`YYYY-MM-DD`), rather than the Foundation concept of a timestamp, I would be expected to use Date.date?

Basically, I think there's some confusion in overloading the Foundation `Date` with something of similar but not equivalent functionality.

I wonder if we want to namespace these, as in:

```swift
enum DateScalars { }

extension DateScalars {
    struct Date { ... }
    struct DateTime { ... }
    struct Time { ... }
}
```

That way there's never a confusion whether I'm expecting a Date or a DateTime. If a customer wants to use `AWSDate`, we can represent it as an Amplify `DateScalar.Date`. More generic implementations can continue to use Foundation `Date` with no special handling.

Thoughts?

Reply:
After a lot of deliberation with the Android team we landed on the following names:

- `Temporal.Date`
- `Temporal.DateTime`
- `Temporal.Time`

Which has the positive side-effect of dropping the `Foundation.Date` extension, the `Temporal.Date` type is completely decoupled from the `Date` type.